### PR TITLE
Update and clean up documentation.

### PR DESCRIPTION
**Pull request by cgsunkel**
## Description of change

As part of the work to decomission the component library I'm checking through the docs to see if there are any references to it anywhere (and using it as an excuse to update anything that's out of date).

I have not looked at the Redux docs (as there is already a work item to rewrite these) or the wiki (this hasn't been updated since 2018 and lots of it is outdated).

There are several areas I'm unsure about:

- I was under the impression that we aren't using Nightwatch any more? It's mentioned [here](https://github.com/uktrade/data-hub-frontend/blob/chore/cleanup-documentation/docs/Folder%20structure.md#acceptance) and some other places.
- There are two sections relating to naming conventions in [this file](https://github.com/uktrade/data-hub-frontend/blob/chore/cleanup-documentation/docs/Sub-apps.md#naming-convention) that are 'TBD', is there anything to go in here or can we delete them?
- I'm not convinced that we need to keep the [contribution guidelines](https://github.com/uktrade/data-hub-frontend/blob/chore/cleanup-documentation/CONTRIBUTING.md) as most of them are duplicated in the [code review guidelines](https://github.com/uktrade/data-hub-frontend/blob/chore/cleanup-documentation/docs/Code%20review%20guidelines.md).
- The code review guidelines haven't been updated for almost a year, is it worth discussing these again (asynchronously)?

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [ ] Has the branch been rebased to master?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or Acceptance)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: docs/Upgrading Node version.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ Engines in `package.json` needs updating:
 - [Image for tests (used by CircleCI)](../test/Dockerfile)
 - [Image for visual tests](../test/visual/Dockerfile)
 
-Once the "Image for tests" has been updated, you will need to [follow the instructions to build and push](Running tests.md) a new version of the image.
+Once the "Image for tests" has been updated, you will need to [follow the instructions to build and push](./Running tests.md) a new version of the image.
```
Comment (cgsunkel) [inline docs/Upgrading Node version.md:26]: I can't get this URL to play nicely

File: README.md [hunk 2]
```diff
@@ -36,69 +36,21 @@ and be provided with a back end server to provide the API, data storage and sear
...
-4. Run `make start-mock`.
-
-5. You can now access the frontend at `http://localhost:3000`.
-
-6. You can check Docker logs by running:
-
-    ```bash
-    `make mock` logs -f
-    ```
+Please view the dedicated [Docker readme](./docs/Docker.md).
```
Comment (cgsunkel) [inline README.md:39]: This was discussed in Slack a few days ago and this was the agreed outcome

File: README.md [hunk 3]
```diff
@@ -36,69 +36,21 @@ and be provided with a back end server to provide the API, data storage and sear
...
+**Note for Civil Servant developers** When running the project natively for the first time on your DIT-issued device you will need to setup ZSH. Instructions for this are available [here](./docs/ZSH%20setup.md)
+
 1.  Navigate to the project root.
 
 2.  Install the required version of node:
 
     ```bash
     brew install nvm
-    nvm use 10.16.0
+    nvm use 12.15.0
```
Comment (ian-leggett) [inline README.md:53]: Ideally we shouldn't have to do this, if we are using nvm then we should have a `.nvmrc` which contains the node version inside so all we would need to do is run `nvm use`

**Review by ian-leggett — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (cgsunkel) [reply README.md:53]: I can pick this up after this is merged

**Review by rachaelcodes — APPROVED** [review]

**Pull request merged**